mlflow/pyfunc/backend.py

```diff
@@ -161,8 +170,31 @@ def predict(self, model_uri, input_path, output_path, content_type):
                 predict_cmd += ["--input-path", shlex.quote(str(input_path))]
             if output_path:
                 predict_cmd += ["--output-path", shlex.quote(str(output_path))]
-            return self.prepare_env(local_path).execute(" ".join(predict_cmd))
+
+            if pip_requirements_override and self._env_manager == _EnvManager.CONDA:
+                # Conda use = instead of == for version pinning
+                pip_requirements_override = [
+                    l.replace("==", "=") for l in pip_requirements_override
+                ]
+
+            environment = self.prepare_env(
+                local_path, pip_requirements_override=pip_requirements_override
+            )
+
+            try:
+                environment.execute(" ".join(predict_cmd))
+            except ShellCommandException as e:
+                raise ShellCommandException(
```

Comment:
Just to show additional messages so that users won't overlook the subprocess error shown above the main process error.